app/services/device_tracking/manage_device.rb

```diff
@@ -0,0 +1,36 @@
+module DeviceTracking
+  class ManageDevice
+    # primary method: updates the usage time if it is not a new device, and calls create
+    # if the device used to log in has not been used before and alerts the user via email
+    # and sms about the new device in use
+    # returns the existing, updated device or the new one just created
+    # :reek:DuplicateMethodCall
+
+    def self.call(user, hash, remote_ip, user_agent)
+      user_has_multiple_devices = UserDecorator.new(user).devices?
+      device = DeviceTracking::LookupDeviceForUser.call(user.id, hash)
+
+      if device
+        DeviceTracking::UpdateDevice.call(device, remote_ip)
+      else
+        new_device = DeviceTracking::CreateDevice.call(user.id, remote_ip, user_agent, hash)
+        alert_user_of_new_device(user, new_device) if user_has_multiple_devices
+        new_device
```

Comment:
If I removed this line all the tests would still pass.  Hence the need for 
'user signs in from a new device in 'manage_device_spec.rb

Reply:
Ah! Ok. Thank you for catching that.

Reply:
Looks like new_device_tracking_spec.rb picks this up.  I stand corrected!